File: lib-segment/src/main/java/websiteschema/mpsegment/tools/accurary/ErrorAnalyzer.java
package websiteschema.mpsegment.tools.accurary;

import websiteschema.mpsegment.core.WordAtom;

import java.util.Map;
import java.util.Set;

public interface ErrorAnalyzer {

    public int getErrorOccurTimes();

    public Set<String> getWords();

    public boolean analysis(WordAtom expect, String possibleErrorWord);

    public void postAnalysis(Map<String, Integer> allWordsAndFreqInCorpus);
}

File: lib-segment/src/main/scala/websiteschema/mpsegment/tools/accurary/ErrorAnalyzer.scala
package websiteschema.mpsegment.tools.accurary

import websiteschema.mpsegment.core.Word

import collection.mutable.Map
import collection.mutable.ListMap

trait ErrorAnalyzer {

    def getErrorOccurTimes() : Int

    def getWords(): ListMap[String,Int]

    def analysis(expect: Word, possibleErrorWord: String) : Boolean

    def postAnalysis(allWordsAndFreqInCorpus: Map[String,Int])
}
